internal/jimm/ssh/ssh.go

```diff
@@ -112,3 +119,42 @@ func (s *sshManager) ControllerInfoFromModelUUID(ctx context.Context, modelUUID
 		JWT:       string(jwt),
 	}, nil
 }
+
+// DialControllerSSHServer determines which controller holds the desired
+// model and dials the controller returning an SSH connection to the controller.
+// We intentionally don't accept user input for the port to disallow the
+// user from probing for open ports on the Juju controller.
+func (s *sshManager) DialControllerSSHServer(ctx context.Context, modelUUID string, user *openfga.User) (*gossh.Client, error) {
```

Comment:
This is doing too much, we can discover the controller separately and pass the connection info. Would also make testing easier. Otherwise rename it to "DiscoverControllerByModelAndDialControllerSSHServer" if you know what I mean?  Cause really this is a side effect that the function name doesn't portray.

Reply:
I don't agree with this one. It's looking up the controller and dialing. If we want the dialer to be something we define outside and pass in, we can do that in a separate PR. Also the docstring states that it finds the controller by the model.

Reply:
Brought the `controllerInfoFromModelUUID` out of the `DialControllerSSHServer` so that it's not doing two things in one.